examples/regression/2-advanced-analysis/plot_coverage_validity.py

```diff
@@ -0,0 +1,528 @@
+        )
+        conformalizer.fit(X_cal, y_cal)
+
+    # Prediction step
+    if isinstance(conformalizer, StandardConformalizer):
+        _, y_pis = conformalizer.predict(X_test)
+    else:
+        _, y_pis = conformalizer.predict(X_test, alpha=1-delta)
+
+    # Coverage step
+    fraction_within_bounds = regression_coverage_score_v2(y_test, y_pis)
+
+    return fraction_within_bounds
+
+
+def run_get_coverage_split(model, params, n_calib, data, target, delta):
+    if not params:
+        ref_reg = StandardConformalizer(model, non_conformity_func, delta)
+        return get_coverage_split(ref_reg, data, target, delta)
+    try:
+        mapie_reg = MapieRegressor(estimator=model, **params(n_calib))
+        coverage = get_coverage_split(mapie_reg, data, target, delta)
+    except Exception:
+        coverage = np.nan
+    return coverage
+
+
+STRATEGIES = {
+    "reference": None,
+    "prefit": lambda n: dict(
+        method="base",
+        cv="prefit",
+        conformity_score=AbsoluteConformityScore(sym=True)
+    ),
+    "prefit_asym": lambda n: dict(
+        method="base",
+        cv="prefit",
+        conformity_score=AbsoluteConformityScore(sym=False)
+    ),
+}
+
+
+##############################################################################
+# Experiment 3: Again but with different MAPIE CP methods
+# -------------------------------------------------------
+#
+# The methods always follow different trajectories but always achieve the
+# expected coverage.
+# Since asymmetric scores can be used, the limits are not exactly the same.
+# We should calculate them differently but that doesn't change our conclusion.
+
+# Parameters of the modelisation
+delta = 0.8
+n_calib = 12  # for asymmetric non-conformity scores
+num_splits = 1000
+
+# Run the experiment
+cumulative_averages_dict = dict()
+
+for method, params in STRATEGIES.items():
+    coverages_list = []
+    run_params = model, params, n_calib, data, target, delta
+    coverages_list = Parallel(n_jobs=-1)(
+        delayed(run_get_coverage_split)(*run_params)
+        for _ in range(num_splits)
+    )
+    cumulative_averages_dict[method] = cumulative_average(coverages_list)
+
+# Plot the results
+fig, ax = plt.subplots()
+for method in STRATEGIES:
+    plt.plot(cumulative_averages_dict[method], alpha=0.5, label=method)
+
+plt.hlines(exact_cov, 0, num_splits, color='r', ls='--', label='Exact Cov.')
+plt.hlines(lower_bound, 0, num_splits, color='k', label='Lower Bound')
+plt.hlines(upper_bound, 0, num_splits, color='b', label='Upper Bound')
+
+plt.xlabel(r'Split Number')
+plt.ylabel(r'$\overline{\mathbb{C}}$')
+plt.title(r'$|D_{cal}| = $' + str(n_calib) + r' and $\delta = $' + str(delta))
+
+plt.legend(loc="upper right", ncol=2)
+plt.ylim(0.7, 1)
+plt.tight_layout()
+plt.show()
+
+
+##############################################################################
+# Experiment 4: Extensive experimentation on different delta and n_calib
```

Comment:
I would only show the `reference`, `prefit`.

Reply:
It is important to show the asymmetric case to avoid any misunderstanding/publication of an issue that is not an issue.